sql/core/src/main/scala/org/apache/spark/sql/json/JsonRDD.scala

```diff
@@ -127,7 +135,7 @@ private[sql] object JsonRDD extends Logging {
       StructType((topLevelFields ++ structFields).sortBy(_.name))
     }
 
-    makeStruct(resolved.keySet.toSeq, Nil)
+    nullTypeToStringType(makeStruct(resolved.keySet.toSeq, Nil))
```

Comment:
I think we should not apply `nullTypeToStringType` at here. Otherwise, we will not know if there is any `NullType` in the inferred schema. Keeping the `NullType` is pretty important when we want to union schemas in future. For example, the type of a field can be `NullType` in one dataset and can be `LongType` in another field. If we eagerly convert `NullType` to `StringType`. After union, we will have `StringType` instead of `LongType`.